dynamic-relations/src/main/java/at/drm/service/DynamicRelationsPrintService.java

```diff
@@ -0,0 +1,127 @@
+package at.drm.service;
+
+import at.drm.exception.NoRelationDaoFoundException;
+import at.drm.model.RelationIdentity;
+import at.drm.model.RelationLink;
+import jakarta.annotation.PostConstruct;
+import jakarta.persistence.EntityManager;
+import java.lang.reflect.Modifier;
+import java.util.ArrayDeque;
+import java.util.Collection;
+import java.util.Deque;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.config.BeanDefinition;
+import org.springframework.context.annotation.ClassPathScanningCandidateComponentProvider;
+import org.springframework.core.type.filter.AssignableTypeFilter;
+import org.springframework.stereotype.Service;
+
+@Service
+@Slf4j
+@RequiredArgsConstructor
+public class DynamicRelationsPrintService {
+
+    private final EntityManager entityManager;
+    private final RelationService relationService;
+
+    private Map<String, Class<RelationIdentity>> classBySimpleName;
+
+    public void printMappedRelations(RelationIdentity relationIdentity) {
```

Comment:
I think the method  `printMappedRelations `should return the generated tree as a `String`.